src/App.js

```diff
@@ -1,20 +1,441 @@
+          // timeTaken: "",
+          // keyTakeaways: "",
+        },
+        // {
+        //   id: 1,
+        //   title: "Fixed software",
+        //   task: "Fixing coding bugs on the software for Machine Learning AI",
+        // },
+      ],
+    };
+  }
+
+  // Transfer tasks from Open Tasks to Tasks In Progress
+  moveTaskOpen = (id) => {
+    // Find the task in state.tasks using the find method
+    const taskToMove = this.state.tasks.find((task) => task.id === id);
+    /** find()
+     * find() method loops through the array elements and returns the first element that is true for task.id === id
+     */
+
+    // if there is a taskToMove
+    if (taskToMove) {
+      const newId = this.state.tasksInProgress.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+      /**
+       * const updatedTask = { ...taskToMove, id: newId };
+       * { ...taskToMove }: ... spread operator creates a shallow copy of the taskToMove object and spreads them into a new object
+       * id: newId modifies the id property of the shallow copy and assignes the value of `newId` to the `id` property of the copied object
+       */
+
+      this.setState(
+        (prevState) => ({
+          // Wrap the arrow function in () to represent object literal and not function block!
+          tasks: prevState.tasks.filter((task) => task.id !== id),
+          tasksInProgress: [...prevState.tasksInProgress, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // Transfer tasks from Tasks In Progress to Open Tasks
+  moveTaskInProgressBacktoOpen = (id) => {
+    const taskToMove = this.state.tasksInProgress.find(
+      (task) => task.id === id
+    ); // finds task matching the id passed into the argument using find() method
+
+    if (taskToMove) {
+      const newId = this.state.tasks.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+
+      this.setState(
+        (prevState) => ({
+          tasksInProgress: prevState.tasksInProgress.filter(
+            (task) => task.id !== id
+          ),
+          tasks: [...prevState.tasks, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // Transfer tasks from In Review back to In Progress
+  moveTaskInReviewBackToInProgress = (id) => {
+    const taskToMove = this.state.tasksInReview.find((task) => task.id === id);
+
+    if (taskToMove) {
+      const newId = this.state.tasksInProgress.length + 1;
+      const updatedTask = { ...taskToMove, id: newId };
+
+      this.setState(
+        (prevState) => ({
+          tasksInReview: prevState.tasksInReview.filter(
+            (task) => task.id !== id
+          ),
+          tasksInProgress: [...prevState.tasksInProgress, updatedTask],
+        }),
+        this.updateLocalStorage
+      );
+    }
+  };
+
+  // Transfer tasks from In Progress to In Review
+  moveTaskInProgress = (id) => {
```

Comment:
I see a lot of repeated functions here. Basically you take a single task, and move it to a certain state. Why not change your data structures to acommodate for it?
You could basically have your state have the different possible statuses:
```
this.state = {
    "OPEN": ...,
    "IN_PROGRESS": ...,
    "DONE": ...
}
```

and then you could have a single function that tells me to which status to assign to:

```moveTask = (id, status) => {
      .....

      this.setState(
            (prevState) => ({
                  [status]: [...prevState[status], updatedTask]
})
}
```

or something along those lines. That way we do not need to hardcode every single status and can program more dynamically

Reply:
I could find the task easily by getting the status via the function argument, no matter which status array it is in.

`const taskToMove = this.state[status].find( ... )`

Reply:
We should be able to identify which list we are clicking on a button now to move the items, that way we can pass that argument into the updater functions, and then select the array dynamically